audbackend/core/backend.py

```diff
@@ -272,7 +298,17 @@ def path(
                 f"allowed characters are '{BACKEND_ALLOWED_CHARS}'."
             )
         folder, file = self.split(path)
-        name, ext = os.path.splitext(file)
+        if ext is None:
+            name, ext = os.path.splitext(file)
+        else:
+            if not ext.startswith('.'):
+                ext = '.' + ext
+            if not path.endswith(ext):
+                raise ValueError(
+                    f"Invalid path name '{path}', "
+                    f"does not end on '{ext}'."
+                )
+            name = file[:-len(ext)]
```

Comment:
Before we implement it here, should we maybe add it to [audeer.file_extension()](https://audeering.github.io/audeer/api-audeer.html#file-extension) and reuse it here?

Reply:
But I just realized that it doesn't make much sense to add `ext=` to `audeer.file_extension()`. If you know the extension, why should you use that function. We could add an argument where you specify the number of dots in your file extension, but I'm also not much in favor of it. So for now I would propose to just stick with your implementation here.